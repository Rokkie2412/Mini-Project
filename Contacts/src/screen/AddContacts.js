import React,{useEffect, useState} from 'react';
import {View, Text, Pressable,TextInput,Modal,Image} from 'react-native'
import styles from '../styles/AddContactsStyles'
import Ion from 'react-native-vector-icons/Ionicons'
import { allowAddTrueFalse,addContactToAPI} from '../functions/AddContactFunction';
import ModalAddContact from '../components/ModalPickImage';

const AddContacts = (props) =>{
    const [First,setFirst] = useState('')
    const [Last,setLast] = useState('')
    const [Age,setAge] = useState('')
    const [Photo,setPhoto] = useState('N/A')
    const [allowAdd,setAllowAdd] = useState(true)
    const [modalVisible,setModalVisible] = useState(false)

    return(
        <View style={styles.mainContainer}>
            <View style={styles.HeaderContainer}>
                <Pressable
                onPress={()=>props.navigation.navigate('Home')}
                ><Ion name='arrow-back' style={styles.HeaderBack}/></Pressable>
                <Text style={styles.Header}>Add Contact</Text>
                <Pressable
                disabled={allowAdd}
                onPress={()=>{
                    console.log('Not Disable')
                    addContactToAPI(First,Last,Age,Photo)
                    props.navigation.navigate('Home')
                }}
                ><Ion name='checkmark-sharp' style={styles.HeaderComplete}/></Pressable>
            </View>
            <View style={styles.contactPhotoContainer}>
                <Pressable
                onPress={()=>{setModalVisible(true)}}
                >
                    {Photo === 'N/A' ? <Ion name='person' style={styles.personIcon}/> :
                    <Image source={{uri:Photo}} style={styles.image}/>
                    }
                </Pressable>
            </View>
            <ModalAddContact visible={modalVisible} setvisible={setModalVisible} setFlasePhoto={setPhoto} falsePhoto={Photo}/>
            <View style={styles.nameContainer}>
                <TextInput
                    autoCapitalize="none"
                    autoCorrect={false}
                    style={styles.name}
                    placeholder="First Name"
                    value={First}
                    onChangeText={(tempFirst)=>{
                        setFirst(tempFirst)
                        allowAddTrueFalse(First,Last,Age,allowAdd,setAllowAdd)
                    }}
                />
                <TextInput
                    autoCapitalize="none"
                    autoCorrect={false}
                    style={styles.name}
                    placeholder="Last Name"
                    value={Last}
                    onChangeText={(tempLast)=>{
                        setLast(tempLast)
                        allowAddTrueFalse(First,Last,Age,allowAdd,setAllowAdd)
                    }}
                />
            </View>
            <View style={styles.ageContainer}>
                <TextInput
                    autoCapitalize="none"
                    autoCorrect={false}
                    keyboardType='numeric'
                    style={styles.age}
                    placeholder="Age"
                    value={Age}
                    onChangeText={(tempAge)=>{
                        allowAddTrueFalse(First,Last,Age,allowAdd,setAllowAdd)
                        setAge(tempAge)  
                    }}
                />
            </View>
            {First==='' || Last ==='' || Age ===''?<Text style={styles.errorvalidation}>*You must fullfill all forms</Text>:
            <Text></Text>}
            
        </View>
    )
}

export default AddContacts