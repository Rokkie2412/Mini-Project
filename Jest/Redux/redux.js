export const ADD = 'ADD'
export const DELETE = 'DELETE'

export const addArray = (data) =>({
    type:ADD,
    payload:data,
})

export const deleteFromArray = (title) => ({
    type:DELETE,
    payload:title,
})

let initialState = {
    arr:[]
}

export const mainReducer = (state=initialState,action) => {
    switch(action.type){
        case ADD:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
        case DELETE:
            return{
                ...state,arr:state.arr.filter(arr => arr.id !== action.payload)
            }
        default:
            return state
    }
}

export default mainReducer