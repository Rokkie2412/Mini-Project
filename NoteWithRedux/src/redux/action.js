import { ADDARRAY } from "./actionType";

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})