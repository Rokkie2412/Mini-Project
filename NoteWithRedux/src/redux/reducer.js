import { ADDARRAY } from "./actionType";

const initialState = {
    arr:[],
    
};

export const mainReducer = (state = initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return{
                ...state,arr:[...state.arr,action.payload]
            }
           
        default:
            return state;
    }
}