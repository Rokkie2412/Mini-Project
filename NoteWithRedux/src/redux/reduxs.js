import remove from "lodash.remove"
export const ADDARRAY = 'ADDARRAY'
export const DELETENOTE = 'DELETENOTE'
export const EDITNOTE = 'EDITNOTE'

export const addArray = (data) => ({
    type:ADDARRAY,
    payload:data
})

export const deleteFromArray = (title) => ({
    type:DELETENOTE,
    payload:title
})
export const EditNote = (title) => ({
    type:EDITNOTE,
    payload:title,
})



const initialState = []

export const mainReducer = (state=initialState,action) => {
    switch (action.type) {
        case ADDARRAY:
            return[
                ...state,action.payload
            ]
        case DELETENOTE:
            const deleteNewArray = remove(state, obj => {
                return obj.title != action.payload
            })
            return deleteNewArray
        case EDITNOTE:
            // const filterDelete = map(state,obj =>{
            //     return obj.title == action.payload
            // })
            const filtered = state.filter((state,obj)=>{
                if(obj.title === action.payload){
                    var tamp = obj.title
                    console.log(tamp);
                }
            })
        default:
            return state;
    }
}

export default mainReducer