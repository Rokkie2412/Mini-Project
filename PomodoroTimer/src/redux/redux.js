
/**
 * @format
 * @flow
 */
export const EXCHAHNGETOSECONDS : string = 'EXCHANGETOSECOND'
export const EXCHAHNGETOSECONDBREAK :string = 'EXCHANGETOSECONDBREAK'
export const EXCHAHNGETOSECONDLONGBREAK :string = 'EXCHANGETOSECONDLONGBREAK'
export const GETDATA:string = 'GETDATA'

export const exchangetosecwork = (hour:string,minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDS,
    payloadhourwork : hour,
    payloadminutework : minute,
    payloadsecondwork : second
})

export const exchangetosecBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDBREAK,
    payloadminutebreak : minute,
    payloadsecondbreak : second
})

export const exchangetosecLongBreak = (minute:string,second:string):Object =>({
    type:EXCHAHNGETOSECONDLONGBREAK,
    payloadminutelongbreak : minute,
    payloadsecondlongbreak : second
})

let initialState : Object  ={
    resultWorkinSecond : 0,
    resultBreakinSecond : 0,
    resultLongBreakinSecond : 0
}

export const mainReducer = (state:Object=initialState, action:function):Function =>{
    switch (action.type) {
        case EXCHAHNGETOSECONDS:
            
            const hourTosecond : number = action.payloadhourwork * 3600
            const minuteTosecond : number = action.payloadminutework * 60
            const res : number = hourTosecond+minuteTosecond+action.payloadsecondwork
            return {
                ...state,
                resultWorkinSecond:res
            }
        case EXCHAHNGETOSECONDBREAK:
            const minuteTosecondBreak : number= action.payloadminutebreak * 60
            const resBreak : number = minuteTosecondBreak+action.payloadsecondbreak
            return {
                ...state,
                resultBreakinSecond:resBreak
            }
        case EXCHAHNGETOSECONDLONGBREAK:
            const minuteToSecondLongBreak : number= action.payloadminutelongbreak * 60
            const resLongBreak : number = minuteToSecondLongBreak+action.payloadsecondlongbreak
            return{
                ...state,
                resultLongBreakinSecond:resLongBreak
            }
        default:
            return state
    }
}

export default mainReducer