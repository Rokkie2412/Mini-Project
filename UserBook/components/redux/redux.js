export const ADD_CONTACT = 'ADD_CONTACT'
export const SHOW_CONTACT = 'SHOW_CONTACT'
export const EDITCONTACT = 'EDITCONTACT'

const ApiUrl = 'https://simple-contact-crud.herokuapp.com/contact'


export const addContact = (firstName,lastName,age,image) =>({
    type:ADD_CONTACT,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

export const editContact = (itemId,firstName,lastName,age,image) => ({
    type:EDITCONTACT,
    payloadID : itemId,
    payloadFirstname : firstName,
    payloadLastName : lastName,
    payAge : age,
    payloadImage : image
})

let initialState = {
    arr:[],
}

export const mainReducer = (state=initialState, action) => {
    switch (action.type) {
        case ADD_CONTACT:
            const postContact = () =>{
                fetch('https://simple-contact-crud.herokuapp.com/contact',{
                    method:'POST',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(postContact())
        case EDITCONTACT :
            const editContact = () =>{
                const tempid = action.payloadID 
                
                fetch(`https://simple-contact-crud.herokuapp.com/contact/${tempid}`,{
                    method:'PUT',
                    body: JSON.stringify({
                        firstName : action.payloadFirstname,
                        lastName : action.payloadLastName,
                        age : action.payAge,
                        photo : action.payloadImage
                    }),
                    headers:{
                        'Content-Type' : 'application/json',
                    }
                 }).then(Response=>Response.json()).then(json=>console.log(json))
            }
            return(editContact())
        default:
            return state
    }
}