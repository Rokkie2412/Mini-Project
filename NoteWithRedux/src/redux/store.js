import { createStore } from "redux";
import {mainReducer} from "./reduxs";

export const store = createStore(mainReducer)