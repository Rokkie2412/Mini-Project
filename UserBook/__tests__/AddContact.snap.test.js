import React from "react";
import AddContacts from "../components/screens/AddContact";
import { Provider } from "react-redux";
import { mainReducer } from "../components/redux/redux";
import { createStore } from "redux";
import { create,act } from "react-test-renderer";

jest.mock(
  'react-native-vector-icons/Ionicons',
  () => 'MockedIonicons',
);

jest.mock(
    'react-native-flash-message',
    ()=> 'Mockedreact-native-flash-message'
)

jest.mock(
    'react-native-image-picker',
    ()=> 'Mocked react-native-image-picker'
)

jest.mock(
    'react-native-textinput-effects',
    ()=>'Mocked react native effect input'
)

const navigation={
    navigate:jest.fn()
}

const store = createStore(mainReducer,{status:'default status'})
const tree = create(
    <Provider store={store}>
        <AddContacts navigation={navigation}/>
    </Provider>
)

test('Snap test Home pages',()=>{
    expect(tree).toMatchSnapshot()
})

