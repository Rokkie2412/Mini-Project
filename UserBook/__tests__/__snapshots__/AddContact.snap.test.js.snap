// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snap test Home pages 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "margin": 5,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="backButton"
    >
      <MockedIonicons
        name="chevron-back-outline"
        style={
          Object {
            "borderRadius": 40,
            "color": "#2C3333",
            "fontSize": 30,
            "padding": 10,
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#2C3333",
          "fontSize": 20,
          "fontWeight": "600",
          "padding": 10,
        }
      }
    >
      Save to contact
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="submitButton"
    >
      <MockedIonicons
        name="checkmark-sharp"
        style={
          Object {
            "borderRadius": 40,
            "color": "#2C3333",
            "fontSize": 30,
            "padding": 10,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignContent": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <MockedIonicons
        name="person"
        style={
          Object {
            "backgroundColor": "#EEEEEF",
            "borderRadius": 100,
            "color": "white",
            "fontSize": 40,
            "padding": 20,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Input Last Name"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "color": "#2C3333",
          "fontWeight": "bold",
          "marginHorizontal": 35,
          "marginTop": 8,
          "paddingLeft": 20,
        }
      }
      value=""
    />
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      onChangeText={[Function]}
      placeholder="Input Last Name"
      style={
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "color": "#2C3333",
          "fontWeight": "bold",
          "marginHorizontal": 35,
          "marginTop": 8,
          "paddingLeft": 20,
        }
      }
      value=""
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 11,
          "marginHorizontal": 32,
          "marginVertical": 7,
        }
      }
      testID="textalert"
    >
      
    </Text>
    <TextInput
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Age?"
      style={
        Object {
          "alignSelf": "center",
          "borderBottomWidth": 1,
          "borderColor": "#333",
          "fontWeight": "bold",
          "marginHorizontal": 12,
          "marginTop": 8,
          "textAlign": "center",
          "width": 100,
        }
      }
      value=""
    />
  </View>
</View>
`;
